packages/framework/src/Command/Elasticsearch/AbstractElasticsearchIndexCommand.php

```diff
@@ -82,14 +91,27 @@ private function getAffectedIndexes(?string $indexName): array
     /**
      * @param \Symfony\Component\Console\Output\OutputInterface $output
      * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex $index
+     * @param int|null $domainId
      */
-    protected function executeForIndex(OutputInterface $output, AbstractIndex $index): void
-    {
-        foreach ($this->domain->getAll() as $domainConfig) {
+    protected function executeForIndex(
+        OutputInterface $output,
+        AbstractIndex $index,
+        ?int $domainId = null,
+    ): void {
+        if ($domainId !== null) {
+            $domainConfig = $this->domain->getDomainConfigById($domainId);
+
             $this->executeCommand(
                 $this->indexDefinitionLoader->getIndexDefinition($index::getName(), $domainConfig->getId()),
```

Comment:
When this is run on the export or export changed command, only one domain will be exported, but the marking to export will be removed for all domains and the information about the need to export will be lost.